int8[] dim2
